x/ecocredit/core/msg_mint_batch_credits.go

```diff
@@ -99,5 +96,11 @@ func validateOriginTx(o *OriginTx, required bool) error {
 	if !reOriginTxSource.MatchString(o.Source) {
 		return errBadReq.Wrap("origin_tx.source must be at most 32 characters long, valid characters: alpha-numberic, space, '-' or '_'")
 	}
+	if len(o.Contract) > 0 && !isValidEthereumAddress(o.Contract) {
+		return sdkerrors.ErrInvalidAddress.Wrapf("%s is not a valid ethereum address", o.Contract)
+	}
```

Comment:
Was not sure whether we should always require a valid ethereum address when validating origin tx and required is set to true in the validation function. Dynamic minting could have other use cases.